bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/airqHandler.java

```diff
@@ -0,0 +1,956 @@
+                        }
+                        logger.trace("air-Q - airqHandler - processType(): property array {} set to {}", channelName,
+                                arrstr.substring(0, arrstr.length() - 2));
+                        getThing().setProperty(channelName, arrstr.substring(0, arrstr.length() - 2));
+                    } else {
+                        logger.error("air-Q - airqHandler - processType(): cannot handle this as an array: {}",
+                                proparr);
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    private void changeSettings(JsonObject jsonchange) {
+        String jsoncmd = jsonchange.toString();
+        logger.trace("air-Q - airqHandler - changeSettings(): called with jsoncmd={}", jsoncmd);
+        if ((!ipaddress.equals("")) && (!password.equals(""))) {
+            Result res = null;
+            try {
+                String url = "http://".concat(ipaddress.concat("/config"));
+                String jsonbody = encrypt(jsoncmd.getBytes(), (String) (getThing().getConfiguration().get("password")));
+                String fullbody = "request=".concat(jsonbody);
+                // String testdecode = decrypt(jsonbody.getBytes(),
+                // (String) (getThing().getConfiguration().get("password")));
+                // logger.trace("air-Q - airqHandler - changeSettings(): testdecode={}, ", testdecode);
+                logger.trace("air-Q - airqHandler - changeSettings(): doing call to url={}, method=POST, body={}", url,
+                        fullbody);
+                res = getData(url, "POST", fullbody);
+                if (res != null) {
+                    Gson gson = new Gson();
+                    JsonElement ans = gson.fromJson(res.getBody(), JsonElement.class);
+                    if (ans != null) {
+                        JsonObject jsonObj = ans.getAsJsonObject();
+                        String jsonAnswer = decrypt(jsonObj.get("content").getAsString().getBytes(),
+                                (String) (getThing().getConfiguration().get("password")));
+                        logger.trace("air-Q - airqHandler - changeSettings(): call returned {}", jsonAnswer);
+                    } else {
+                        logger.error("The air-Q data could not be extracted from this string: {}", ans);
+                    }
+                }
+            } catch (Exception e) {
+                System.out.println(
+                        "air-Q - airqHandler - prepareChangeSettings(): Error while changing settings in air-Q data: "
+                                + e.toString());
+            }
+        }
+    }
+    // Getting the data files will be implemented in future, but it is not ready yet
+    /*
+     * private void getDataFiles() {
+     * Result res = null;
+     * String url = "http://".concat(ipaddress.concat("/dirbuff"));
```

Comment:
The norm is to remove all commented out code as it will create a lot of lines that change when the comment block is removed.